rust/lance/src/index/vector/utils.rs

```diff
@@ -65,3 +70,53 @@ pub async fn maybe_sample_training_data(
     })?;
     Ok(array.as_fixed_size_list().clone())
 }
+
+#[derive(Debug)]
+pub struct PartitionLoadLock {
```

Comment:
Is this `pub(crate)`. Just wanted to make sure this is not public API

Reply:
this is in a private module, so it won't be part of public API